code/game/objects/structures/flora.dm

```diff
@@ -185,12 +185,19 @@ GLOBAL_LIST_EMPTY(flora_uprooting_tools_typepaths)
 	. = FALSE
 	if(harvested || !harvestable)
 		return null
-	
+
 	if(harvesting_item)
-		if(!is_type_in_typecache(harvesting_item, required_tools))
-			return
+		//Check if its disallowed first, because we wanna cut it down in its tracks if so
 		if(is_type_in_typecache(harvesting_item, disallowed_tools))
 			return
+		if(!is_type_in_typecache(harvesting_item, required_tools))
+			//Check to see if wooden flora is being attacked by a saw item (letting the items on/off state control this is better than putting them in the list)
+			if((flora_flags & FLORA_WOODEN) && (harvesting_item.item_flags & TOOL_SAW))
+				return TRUE
+			//Check to see if stone flora is being attacked by a mining item (same reason as above)
+			if((flora_flags && FLORA_STONE) && (harvesting_item.item_flags && TOOL_MINING))
```

Comment:
```suggestion
			if((flora_flags & FLORA_STONE) && (harvesting_item.item_flags & TOOL_MINING))
```

Reply:
doip, it should be harvesting_item.tool_behaviour == too